bitseq/sequence.go

```diff
@@ -355,22 +360,25 @@ func (h *Handle) set(ordinal, start, end uint64, any bool, release bool, serial
 			nh.unselected--
 		}
 
-		// Attempt to write private copy to store
-		if err := nh.writeToStore(); err != nil {
-			if _, ok := err.(types.RetryError); !ok {
-				return ret, fmt.Errorf("internal failure while setting the bit: %v", err)
+		if h.store != nil {
```

Comment:
This is to differentiate between a store logic and a non store logic codepath. I need to unlock before going to write store it will lead to deadlock.

Reply:
nh.writeStore() takes the datastore lock. So in this scenario we take in h.lock followed by ds.lock (part of putObjectAtomic).

Now at the start of the set function we take the ds.lock (part of getObject) and then take h.lock() (also part of getObject ->> cache.get() ) So this will cause 2 go routines to go into a deadlock where go1 will take ds.lock and wait on h.lock and go2 will take in h.lock and wait on ds.lock

Reply:
Got it regarding the logic to differentiate store/non-store codepaths. 

For the locks, it looks like we are calling `nh.writeToStore()` below which will lock `nh`'s mutex and not that of `h`, right?

Reply:
Yep thats right. All the more reason not to have a lock around the operation right ? I retained the logic needed for store code path the way it was.